ldap_auth_provider.py

```diff
@@ -401,27 +393,24 @@ def _ldap_simple_bind(self, server, bind_dn, password):
             raise
 
     @defer.inlineCallbacks
-    def _ldap_authenticated_search(self, server, password, filters, attributes=[]):
-        """ Attempt to login with the preconfigured bind_dn
-            and then continue searching and filtering within
-            the base_dn
+    def _ldap_authenticated_search(self, server, password, filters):
+        """Attempt to login with the preconfigured bind_dn and then continue
+        searching and filtering within the base_dn.
+
+        Fetches the attributes that correspond to uid/name/mail as defined in
+        the config.
 
+        Args:
             server (str): The LDAP server to connect to.
             password (str): The user's password.
             filters (List[Tuple[str,str]]): A list of tuples of key/value
                 pairs to filter the LDAP search by.
-            attributes (List[str]): A list of strings of attribute names to
-                return.
-
-            Returns (True, LDAP3Connection)
-                if a single matching DN within the base was found
-                that matched the filter expression, and with which
-                a successful bind was achieved
 
-                The LDAP3Connection returned is the instance that was used to
-                verify the password not the one using the configured bind_dn.
-            Returns (False, None)
-                if an error occured
+        Returns:
+            Deferred[tuple[bool, LDAP3Connection, response]]: Returns a 3-tuple
+            where first field is whether a *single* entry was found, the second
+            is the open connection bound to the found user and the final field
```

Comment:
it'd be nice to say that the second and third  fields are `None` and `None` if the first is `false`.

(which makes one wonder what the point of the first field is: why not just check for `None`ness of the entry? But that sounds like a problem for another day)